sql/catalyst/src/test/java/org/apache/spark/sql/catalyst/expressions/RowBasedKeyValueBatchSuite.java

```diff
@@ -338,15 +338,18 @@ public void appendRowUntilExceedingCapacity() throws Exception {
 
   @Test
   public void appendRowUntilExceedingPageSize() throws Exception {
+    // Use of this property prevents issues when running on a two core machine
+    // A developer may override this size to 1 MB not 64 MB, so use that if so
+    int pageSizeToUse = Integer.parseInt(System.getProperty("spark.buffer.pageSize"));
```

Comment:
Yeah that's right, couldn't figure out how to get the SparkConf from that test though, will have another look now